src/validation/statement.rs

```diff
@@ -309,9 +326,16 @@ fn validate_cast_literal<T: AnnotationMap>(
     );
 
     if !literal.is_cast_prefix_eligible() {
-        validator.push_diagnostic(Diagnostic::literal_expected(location.clone()))
+        validator.push_diagnostic(
+            Diagnostic::error(format!(
+                "Cannot cast into {}, only elementary types are allowed",
```

Comment:
Maybe we can add a backtick here, as we use it quite often in other places i.e. "Cannot cast into `{}` ..."